fathom_fox/addon/vector.js

```diff
@@ -137,6 +137,16 @@ class Vectorizer extends PageVisitor {
     }
 
     async processAtEndOfRun() {
+        // Remove potential duplicated feature vectors from the vector file.
+        // 7/24/2020: This is a bandage solution to the problem of duplicated
+        // feature vectors. I (Daniel) will fix the source of the problem in
+        // ~2-4 weeks. If you are reading this message after that time has
+        // passed, please tell me to fix the problem and get rid of this code.
+        let filenamesSeen = new Set();
+        this.vectors = this.vectors.filter(item => {
+            return filenamesSeen.has(item['filename']) ? false : filenamesSeen.add(item['filename']);
```

Comment:
Looks good, but we can tersen like this:

```js
this.vectors = this.vectors.filter(item => filenamesSeen.has(item['filename']) ? false : filenamesSeen.add(item['filename']));
```

I like your clever use of the set, returned from `add`, as a true value. Even empty sets are true in JS. :-D

Reply:
It occurs to me that de-duping by filename will spuriously remove things if a filename occurs multiple times in the case of nested-folder corpora. But we don't do that much. And we have a ticket open to put in longer pathnames.

Reply:
Do you care about line length here? The form you suggest goes to 132 characters. And you are right this would cause problems with reoccurring filename. We could solve that by looking at the contents of `nodes` in addition to the filename. I had originally just thrown the entire array into a set, but forgot that the timing portion for each file will be different.

Reply:
I'll take comprehensibility and tokenwise brevity every time over short lines.

Since apparently the visitor is really visiting the duplicate pages more than one (the timings being different), I suppose the nodes could be different as well, them being occasionally nondeterministic. Filename should be fine for a quick patch.

Reply:
If you want, you can parenthesize to wrap the lines.